internal/services/automanage/arc_machine_automanage_configuration_assignment_resource.go

```diff
@@ -0,0 +1,165 @@
+package automanage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automanage/2022-05-04/configurationprofileassignments"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automanage/2022-05-04/configurationprofilehcrpassignments"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automanage/2022-05-04/configurationprofiles"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ArcMachineConfigurationAssignment struct {
+	ArcMachineId    string `tfschema:"arc_machine_id"`
+	ConfigurationId string `tfschema:"configuration_id"`
+}
+
+func (v ArcMachineConfigurationAssignment) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"arc_machine_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: configurationprofileassignments.ValidateMachineID,
+		},
+		"configuration_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: configurationprofiles.ValidateConfigurationProfileID,
+		},
+	}
+}
+
+func (v ArcMachineConfigurationAssignment) Attributes() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (v ArcMachineConfigurationAssignment) ModelObject() interface{} {
+	return &ArcMachineConfigurationAssignment{}
+}
+
+func (v ArcMachineConfigurationAssignment) ResourceType() string {
+	return "azurerm_arc_machine_automanage_configuration_assignment"
+}
+
+func (v ArcMachineConfigurationAssignment) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Automanage.ConfigurationProfileArcMachineAssignmentsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ArcMachineConfigurationAssignment
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			arcMachineId, err := configurationprofileassignments.ParseMachineID(model.ArcMachineId)
+			if err != nil {
+				return err
+			}
+
+			configurationId, err := configurationprofiles.ParseConfigurationProfileID(model.ConfigurationId)
+			if err != nil {
+				return err
+			}
+
+			// Currently, the configuration profile assignment name has to be hardcoded to "default" by API requirement.
+			id := configurationprofilehcrpassignments.NewProviders2ConfigurationProfileAssignmentID(subscriptionId, arcMachineId.ResourceGroupName, arcMachineId.MachineName, "default")
```

Comment:
is there a plan to change this in the future? should we add a property for this with the value "default" defaulted and only allowing default so its clear this is what the resource is working wioth + future proof?

Reply:
As far as I'm aware, there are no plans to change this in the future. It seems that the linking of resource 1 with resource 2 is fixed to a single option, which is why it's set as the "default". Introducing a new property appears unnecessary to me, as it would essentially just an alias for "default". WDYT?